src/services/bria/index.ts

```diff
@@ -125,7 +132,38 @@ export const NewOnChainService = (): INewOnChainService => {
   const metadata = new Metadata()
   metadata.set("x-bria-api-key", BRIA_PROFILE_API_KEY)
 
-  return {}
+  const createOnChainAddress = async (
+    requestId?: OnChainAddressRequestId,
+  ): Promise<OnChainAddressIdentifier | UnknownOnChainServiceError> => {
+    try {
+      const request = new NewAddressRequest()
+      request.setWalletName(BRIA_WALLET_NAME)
+      if (requestId !== undefined) {
+        request.setExternalId(requestId)
+      }
+
+      const newAddressWithMetadataOverload = (
+        { request, metadata }: { request: NewAddressRequest; metadata: Metadata },
+        callback: (error: ServiceError | null, response: NewAddressResponse) => void,
+      ) => {
+        const newAddress = bitcoinBridgeClient.newAddress.bind(bitcoinBridgeClient)
+        return newAddress(request, metadata, callback)
+      }
```

Comment:
why to use this structure if you will promisify later?

Reply:
It's because promisify expects 2 arguments where the 2nd one is a callback, so I had to collect all the arguments from `newAddress` before the callback argument into a single one to pass on to promisify

Reply:
yes, what I mean is that you dont need to prmisify, just create an async function

Reply:
Don't we need promisify still to make the callback functionality into a promise?

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/2662/commits/8f0a19a4d2ac11aae61fdb5a6f1eb66be95473af